app/source/js/activities/serviceProfessionalServiceEditor.js

```diff
@@ -63,13 +65,19 @@ A.prototype.show = function show(options) {
     this.viewModel.pricingType(null);
 
     // Params
-    var params = options && options.route && options.route.segments || [];
+    var matcher = new RouteMatcher([
+        new Route(':jobTitleID/pricing_type/:pricingTypeID/client/:clientID/new'),
+        new Route(':jobTitleID/pricing_type/:pricingTypeID/new'),
+        new Route(':jobTitleID/:serviceID')
+    ]);
 
-    var jobTitleID = params[0] |0,
-        // Parameter [1] can be 'new' followed by a pricingTypeID as [2]
-        pricingTypeID = params[1] === 'new' ? params[2] |0 : 0,
-        // Or a pricingID
-        serviceProfessionalServiceID = params[1] |0;
+    var paramsDefaults = { jobTitleID: 0, serviceID: 0, pricingTypeID: 0, clientID: 0 },
+        params = matcher.match(options.route.segments.join('/'), paramsDefaults) || {};
+
+    var jobTitleID = +params.jobTitleID,
```

Comment:
Just to note that I use the expression "|0" rather than the plus operator when I get an integer value for a parameter (as in the code before modification) because it ensures to *ever* gets a valid integer number as result (in case you don't add or don't need to apply validation, as in this case); using the plus operator you may end with the special value NaN if the original value is *undefined*, a non-empty string, or any type or object different than a number; in the other hand, if you need to retain the decimal part, the "|0" cannot be used.

It can lead to bugs if in the code the value is compared against 0, for example " if (jobTitleID == 0) " will be false when NaN. But " if (!jobTitleID) " would be true both for NaN and 0. Not in this file, but in other activities and viewmodels we do checks like this one.

Just try in a js scratchpad or console:
```js
[
    [
        +undefined,
        +{},
        +'theid',
        +'',
        +null,
        +false,
        +true,
        +2.2,
        +'3.3',
        +'-4'
    ],
    [
        undefined |0,
        {}
        |0,
        'theid' |0,
        '' |0,
        null |0,
        false |0,
        true |0,
        2.2 |0,
        '3.3' |0,
        '-4' |0
    ]
]
/* In Firefox:
NaN,NaN,NaN,0,0,0,1,2.2,3.3,-4,0,0,0,0,0,0,1,2,3,-4
*/
```